crates/e2e/tests/e2e/setup/onchain_components.rs

```diff
@@ -170,6 +185,76 @@ impl Deref for MintableToken {
     }
 }
 
+#[derive(Debug)]
+pub struct CowToken {
+    contract: CowProtocolToken,
+    holder: Account,
+}
+
+impl CowToken {
+    pub async fn fund(&self, to: H160, amount: U256) {
+        tx!(self.holder, self.contract.transfer(to, amount));
+    }
+
+    pub async fn permit(&self, owner: &TestAccount, spender: H160, value: U256) -> InteractionData {
```

Comment:
Is there a reason these functions are specific to the cow token and can not be generalized in some way?

Reply:
So these are E2E test setup-specific functions for the CoW token. This is because the CoW token uses the same bytecode as on mainnet and (obviously) can't mint tokens arbitrarily to different addresses. The way the CoW token is setup is that it gets created and the entire initial total supply gets allocated to an address (the `self.holder`). Then the holder can distribute these tokens with the `fund` method for unit tests.

`permit` is only supported by the CoW token in our E2E tests (our other ERC20 token implementations do not support that method).